src/sinks/util/rusoto.rs

```diff
@@ -63,7 +63,14 @@ impl AwsCredentialsProvider {
             let creds = AutoRefreshingProvider::new(provider).context(InvalidAWSCredentials)?;
             Ok(Self::Role(creds))
         } else {
-            let creds = DefaultCredentialsProvider::new().context(InvalidAWSCredentials)?;
+            debug!("using default credentials provider for AWS.");
+            let mut chain = ChainProvider::new();
+            // 8 seconds because our default healthcheck timeout
+            // is 10 seconds.
+            chain.set_timeout(Duration::from_secs(8));
+
+            let creds = AutoRefreshingProvider::new(chain)?;
```

Comment:
Should this have `.context(InvalidAWSCredentials)`? If not, where can the `InvalidAWSCredentials` error happen in this flow?

Reply:
Oh its missing this, I totally missed that from the diff! Let me add that back.